### PR TITLE
[WIP] api: add relocate region admin operator.

### PR DESCRIPTION
**Pull request by disksing**
For fast disperse small regions before import data.

### CODE REVIEW

**Comment by siddontang** [comment]
should we add a range to open relocate regions in the range? 

I think @huachaohuang only want to relocate the regions that interest him.

**Comment by disksing** [comment]
Can't use range since the imported data could have more than 1 range.
The api just relocate 1 region, import tool will have to call api multiple times.

File: server/schedule/selector.go [hunk 1]
```diff
@@ -166,3 +166,41 @@ func (s *randomSelector) SelectTarget(stores []*core.StoreInfo, filters ...Filte
 	}
 	return s.Select(candidates)
 }
+
+// SelectStoresToRelocate selects stores to relocate a region. It randomly
+// selects the target stores while satisfying the constraints of replica
+// placement to the maximum extent.
+func SelectStoresToRelocate(stores []*core.StoreInfo, opt Options, filters ...Filter) []*core.StoreInfo {
```
Comment (siddontang) [inline server/schedule/selector.go]: any test?

**Comment by siddontang** [comment]
ping @disksing

**Comment by disksing** [comment]
@siddontang this branch is still POC status, I'll improve the PR after we verified this approach is practical.

**Comment by huachaohuang** [comment]
@disksing  Thanks, I have implemented another version as a grpc API in https://github.com/pingcap/pd/pull/890.

**Pull request closed without merging**